aws_topology/stackstate_checks/aws_topology/resources/redshift.py

```diff
@@ -8,34 +29,60 @@ class RedshiftCollector(RegisteredResourceCollector):
     COMPONENT_TYPE = "aws.redshift"
     CLOUDFORMATION_TYPE = "AWS::Redshift::Cluster"
 
+    @set_required_access_v2("redshift:DescribeTags")
+    def collect_tags(self, cluster_arn):
+        return [
+            tags
+            for tags in client_array_operation(
+                self.client, "describe_tags", "TaggedResources", ResourceName=cluster_arn, ResourceType="Cluster"
+            )
+        ]
+
+    def collect_cluster(self, cluster_data):
+        cluster_arn = self.agent.create_arn(
+            "AWS::Redshift::Cluster", self.location_info, cluster_data.get("ClusterIdentifier", "UNKNOWN")
+        )
+        tags = self.collect_tags(cluster_arn) or []
+        return ClusterData(cluster=cluster_data, tags=tags, arn=cluster_arn)
+
+    def collect_clusters(self, **kwargs):
+        for cluster in [
+            self.collect_cluster(cluster_data)
+            for cluster_data in client_array_operation(self.client, "describe_clusters", "Clusters", **kwargs)
+        ]:
+            yield cluster
+
+    @set_required_access_v2("redshift:DescribeClusters")
+    def process_clusters(self):
+        for cluster_data in self.collect_clusters():
+            self.process_cluster(cluster_data)
+
     def process_all(self, filter=None):
-        for page in self.client.get_paginator('describe_clusters').paginate():
-            for redshift_data_raw in page.get('Clusters') or []:
-                redshift_data = make_valid_data(redshift_data_raw)
-                self.process_redshift(redshift_data)
-
-    def process_one_cluster(self, clusterid):
-        for page in self.client.get_paginator('describe_clusters').paginate(ClusterIdentifier=clusterid):
-            for redshift_data_raw in page.get('Clusters') or []:
-                redshift_data = make_valid_data(redshift_data_raw)
-                self.process_redshift(redshift_data)
-
-    def process_redshift(self, redshift_data):
-        self.emit_component(redshift_data['ClusterIdentifier'], self.COMPONENT_TYPE, redshift_data)
-        vpcid = redshift_data.get('VpcId')
-        if vpcid:
-            self.emit_relation(redshift_data['ClusterIdentifier'], vpcid, 'uses service', {})
+        if not filter or "clusters" in filter:
+            self.process_clusters()
+
+    def process_one_cluster(self, cluster_id):
+        for cluster_data in self.collect_clusters(ClusterIdentifier=cluster_id):
+            self.process_cluster(cluster_data)
+
+    @transformation()
+    def process_cluster(self, data):
+        cluster = Cluster(data.cluster, strict=False)
+        cluster.validate()
+        output = make_valid_data(data.cluster)
+        output["Name"] = cluster.ClusterIdentifer
+        output["Tags"] = data.tags
+        output["URN"] = [data.arn]
+        self.emit_component(data.arn, self.COMPONENT_TYPE, output)
```

Comment:
So we are changing the ID of this component right? Should we document that in the notion table too?

Reply:
Documented